optimum/onnxruntime/trainer.py

```diff
@@ -788,45 +814,28 @@ def evaluation_loop_ort(
 
             with_loss = has_labels and not self.label_smoother
             self._export(onnx_model_path, with_loss=with_loss, device=export_device)
+
             self.exported_with_loss = with_loss
             self.onnx_model_path = onnx_model_path.as_posix()
             # Fix exported ONNX IR
             fix_atenops_to_gather(self.onnx_model_path)
             logger.info("[INFO] ONNX model is stored in:\n", self.onnx_model_path)
 
-        self.infer_sess = onnxruntime.InferenceSession(
-            self.onnx_model_path,
-            session_options=self.session_options,
-            providers=["CUDAExecutionProvider" if _is_gpu_available() else "CPUExecutionProvider"],
-        )
+        # Load ORT model
+        if not self.exported_with_loss and self.feature in ORTFeaturesManager.SUPPORTED_FEATURES:
+            # Exported with standard outputs, use specific ORTModels
+            ort_model_cls = ORTFeaturesManager.get_model_class_for_feature(self.feature)
+        else:
+            ort_model_cls = ORTModelForCustomTasks
+
+        model_id, file_name = os.path.split(self.onnx_model_path)
+        ort_model = ort_model_cls.from_pretrained(model_id=model_id, file_name=file_name)
 
         args = self.args
 
         prediction_loss_only = prediction_loss_only if prediction_loss_only is not None else args.prediction_loss_only
 
-        # if eval is called w/o train init deepspeed here
-        if args.deepspeed and not self.deepspeed:
-
-            # XXX: eval doesn't have `resume_from_checkpoint` arg but we should be able to do eval
-            # from the checkpoint eventually
-            deepspeed_engine, _, _ = deepspeed_init(
-                self, num_training_steps=0, resume_from_checkpoint=None, inference=True
-            )
-            self.model = deepspeed_engine.module
-            self.model_wrapped = deepspeed_engine
-            self.deepspeed = deepspeed_engine
-
-        model = self._wrap_model(self.model, training=False, dataloader=dataloader)
-
-        # if full fp16 or bf16 eval is wanted and this ``evaluation`` or ``predict`` isn't called
-        # while ``train`` is running, cast it to the right dtype first and then put on device
-        if not self.is_in_train:
-            if args.fp16_full_eval:
-                model = model.to(dtype=torch.float16, device=args.device)
-            elif args.bf16_full_eval:
-                model = model.to(dtype=torch.bfloat16, device=args.device)
-
-        batch_size = self.args.eval_batch_size
```

Comment:
This code snippet is not needed anymore?

Reply:
Hi @regisss,
In the code beyond, for the deepspeed part, as ONNX Runtime doesn't support DeepSpeed stage 3 / infinity yet, it is not necessary to initialize the deepspeed engine during inference. And as for the model weights dtype conversion, as for the inference the model is an ORTModel with an ORT inference session behind, it won't work this way. 
Thanks for pointing that out, indeed currently there are some issues on dtype conversion especially for models trained with mixed-precision. I will add that in the backlog.

Reply:
Thanks for the explanation Jingya :slightly_smiling_face: